### PR TITLE
Items and services CRUD

### PR DESCRIPTION
**Pull request by BenasB**
Create CRUD endpoints for:
- Items
- Services

Also:
- Update generic `preValidation` error formatting
- Add missing 400 documentation on POST and PUT CRUD endpoints for customers

image

### CODE REVIEW

File: src/modules/customers/customer.route.ts
Comment (6ar8nas) [inline src/modules/customers/customer.route.ts:1]: I see I overlooked it myself, but invalid id (non uuid) can also result in a validation error, so 'get by id' and other similar endpoints should also have it

File: src/modules/item/item.controller.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import { RequestByIdParams } from "../generic";
+
+export const getItemsHandler =
+  (server: FastifyInstance) =>
+  async (_request: FastifyRequest, reply: FastifyReply) => {
+    try {
+      const items = await findItems(server);
+      return reply.code(200).send(items);
+    } catch (e) {
+      server.log.error(e);
```
Comment (6ar8nas) [inline src/modules/item/item.controller.ts:20]: I no longer catch errors in other service functions. I guess we should unify the behavior one way or another.

Comment (6ar8nas) [reply src/modules/item/item.controller.ts:20]: Another way may be using onError fastify hook or setErrorHandler I believe

**Review by 6ar8nas — APPROVED** [review]

File: src/modules/item/item.controller.ts (hunk 1 above)
Comment (BenasB) [reply src/modules/item/item.controller.ts]: Made controllers consistent to what is in the main branch

**Pull request merged**